packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -492,20 +492,34 @@ class _CupertinoAlertDialogState extends State<CupertinoAlertDialog> {
   }
 }
 
-/// Rounded rectangle surface that looks like an iOS popup surface, e.g., alert dialog
-/// and action sheet.
+/// An iOS-style rounded rectangle surface useful for showing modal overlays,
+/// such as dialogs and action sheets.
 ///
-/// A [CupertinoPopupSurface] can be configured to paint or not paint a white
-/// color on top of its blurred area. Typical usage should paint white on top
-/// of the blur. However, the white paint can be disabled for the purpose of
-/// rendering divider gaps for a more complicated layout, e.g., [CupertinoAlertDialog].
-/// Additionally, the white paint can be disabled to render a blurred rounded
-/// rectangle without any color (similar to iOS's volume control popup).
+/// By default, this surface is composed of a transparent surface color painted
+/// atop a [BackdropFilter]. The surface color is light gray when the ambient
+/// [CupertinoTheme] brightness is [Brightness.light], and dark gray when the
+/// ambient brightness is [Brightness.dark]. The [BackdropFilter] applied
+/// beneath the surface saturates and blurs underlying content. In other words,
+/// the overall effect is result = add_surface_color(blur(saturate(source))).
+///
+/// The blur strength can be changed by setting [blurSigma] to a positive value,
+/// or removed by setting the [blurSigma] to 0.
+///
+/// The saturation effect can be removed by setting [isVibrancePainted] to
+/// false. The saturation effect is not supported on Web and will not be
```

Comment:
I see it's disabled in the widget by a `kIsWeb` check. Any reason this should not be enabled on the web (instead of letting developer decide what to do)? 

Looks like the reason is the `src` blend mode? The canvaskit renderer supports that I think, so disabling this on web altogether is probably a bit of an overkill (and iirc there are plans to deprecate the HTML renderer)?

Reply:
I confirm that HTML renderer is getting deprecated.

Reply:
On the HTML renderer, I was getting this error: <img width="748" alt="image" src="https://github.com/user-attachments/assets/15354b5f-93bf-4497-bca0-428ddb681b47">

I initially thought it was the ColorFilter, but ImageFilter.compose appears to be the true culprit (I suppose it imports ImageFilter.erode? Not totally sure...)  

To your point, it does work on CanvasKit, so I revised the check to `if ((kIsWeb && !isCanvasKit) || !isVibrancePainted)`. Let me know if I should just check for !isVibrancePainted, or if this works.

Reply:
Have you tried the wasm implementation? I'm wondering if it's supported.

To enable it, use `--wasm`. The framework checks it with `isSkwasm`.

Reply:
The erode thing seems to be a typo:

https://github.com/flutter/engine/blob/6648b3cfe0d0f497cc033bac228751fc9d897ee3/lib/web_ui/lib/src/engine/html/renderer.dart#L139-L146

This is probably supported on wasm too, so `isSkiaWeb` probably.

Reply:
> To your point, it does work on CanvasKit, so I revised the check to if ((kIsWeb && !isCanvasKit) || !isVibrancePainted). Let me know if I should just check for !isVibrancePainted, or if this works.

This SGTM (with `s/isCanvasKit/isSkiaWeb/`). It's likely that the HTML renderer will be deprecated, then people will audit `isSkiaWeb` usages and update the implementation here.

Reply:
Perfect, I subbed for if ((kIsWeb && !isSkiaWeb) || !debugIsVibrancePainted)